dhcpv4/client.go

```diff
@@ -27,10 +27,23 @@ var (
 	DefaultWriteTimeout = 3 * time.Second
 )
 
+// ClientExchangeOptions specifies a set of options to be used for a particular
+// exchange.
+type ClientExchangeOptions struct {
+	// UnicastSocket is used to ask the library to use an unicast socket instead
+	// of a broadcast socket for the DHCPv4 exchange. Note that it's on the
+	// user of the library to set up the actual DHCP packet (ex: by specifying a
+	// server address) so this actually works as expected.
+	UnicastSocket bool
+}
+
 // Client is the object that actually performs the DHCP exchange. It currently
-// only has read and write timeout values.
+// only has read and write timeout values, plus (optional) local and remote
+// addresses.
 type Client struct {
 	ReadTimeout, WriteTimeout time.Duration
+	RemoteAddr   *net.UDPAddr
```

Comment:
I would use net.Addr in order to be more consistent with the v6 client.

Reply:
my fault here! I suggested @gvialetto to use UDPAddr because on top of my mind I thought we were using UDPAddr instead of Addr